Automattic/jetpack
# Jetpack Sync: Add switching logic between Custom table and Options table

This is a spin-off of #31318 and continuation of #32089, #32111 and #32135

## Proposed changes:

This PR builds on top of the migration and initialization logic introduced in #32135 and hooks into the option update for `custom_queue_table_enabled` to either initialize, switch over to the table and migrate the data. 

The PR depends on #32089, #32111 and #32135 and needs to have all of them merged before this one can be merged. We'll also need to update the changelog and package versions afterwards.

To be further described

### Other information:

- [x] Have you written new tests for your changes, if applicable? Tests are in #31318

## Jetpack product discussion

p9dueE-7e6-p2

## Does this pull request change what data or activity we track or use?

No

## Testing instructions:

* Apply PR
* Make sure Jetpack Sync works
* Run the Docker PHPUnit tests - `jetpack docker phpunit` and make sure they pass. You can also run only the Sync unit tests with `jetpack docker phpunit -- --testsuite "sync"`

* All the steps in the testing instructions comment in p9dueE-7e6-p2
